src/o365/spo/commands/mail/mail-send.ts

```diff
@@ -0,0 +1,236 @@
+  body: string;
+  from?: string;
+  cc?: string;
+  bcc?: string;
+  additionalHeaders?: TypedHash<string>;
+}
+
+class SpoMailSendCommand extends SpoCommand {
+  public get name(): string {
+    return commands.MAIL_SEND;
+  }
+
+  public get description(): string {
+    return 'Send an email from SharePoint';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.from = typeof args.options.from !== 'undefined';
+    telemetryProps.cc = typeof args.options.cc !== 'undefined';
+    telemetryProps.bcc = typeof args.options.bcc !== 'undefined';
+    telemetryProps.additionalHeaders = typeof args.options.additionalHeaders !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const resource: string = Auth.getResourceFromUrl(args.options.webUrl);
+    let siteAccessToken: string = '';
+
+    if (this.debug) {
+      cmd.log(`Retrieving access token for ${resource}...`);
+    }
+
+    auth
+      .getAccessToken(resource, auth.service.refreshToken as string, cmd, this.debug)
+      .then((accessToken: string): request.RequestPromise | Promise<void> => {
+        siteAccessToken = accessToken;
+
+        if (this.debug) {
+          cmd.log(`Retrieved access token ${siteAccessToken}.`);
+          cmd.log(``);
+          cmd.log(`auth object:`);
+          cmd.log(auth);
+        }
+
+        const params: any = {};
+        params.properties = {};
+        params.properties.__metadata = { "type": "SP.Utilities.EmailProperties" };
+        params.properties.Body = args.options.body;
+        params.properties.Subject = args.options.subject
+
+        params.properties.To = { results: args.options.to.replace(/\s+/g, '').split(',') };
+        if (this.verbose) {
+          cmd.log(`List of recipients: ${args.options.to}...`);
+        }
+
+        if (args.options.from && args.options.from.length > 0) {
+          params.properties.From = args.options.from;
+          if (this.verbose) {
+            cmd.log(`Mail will send from: ${args.options.from}...`);
+          }
+        }
+
+        if (args.options.cc && args.options.cc.length > 0) {
+          params.properties.CC = { results: args.options.cc.replace(/\s+/g, '').split(',') };
+          if (this.verbose) {
+            cmd.log(`List of addresses to which a carbon copy: ${args.options.cc}...`);
+          }
+        }
+
+        if (args.options.bcc && args.options.bcc.length > 0) {
+          params.properties.BCC = { results: args.options.bcc.replace(/\s+/g, '').split(',') };
+          if (this.verbose) {
+            cmd.log(`List of addresses that receive a copy of the mail but are not listed as recipients: ${args.options.bcc}...`);
+          }
+        }
+
+        if (args.options.additionalHeaders) {
+          params.properties.AdditionalHeaders = args.options.additionalHeaders;
```

Comment:
In the list of options, you set the type of additionalHeaders as TypedHash<string>, but in bash, you can only pass a string. If you pass it as-is, `params.properties.AdditionalHeaders` will get assigned a string rather than a collection. Have you verified that the current approach is actually working?